File: Greedy/Cpp/1946.cpp
// 신입사원
#include <algorithm>
#include <iostream>
#include <vector>
#define P pair<int, int>
using namespace std;

bool cmp(P& a, P& b) {
    if (a.first == b.first) {
        return a.second > b.second;
    } else {
        return a.first < b.first;
    }
}

void f() {
    int n;
    cin >> n;
    vector<P> v;
    for (int i = 0; i < n; i++) {
        int a, b;
        cin >> a >> b;
        v.push_back(make_pair(a, b));
    }
    sort(v.begin(), v.end(), cmp);

    int cri = v[0].second;
    int cnt = 0;
    for (int i = 1; i < n; i++) {
        if (cri >= v[i].second) {
            cnt++;
            cri = v[i].second;
        }
    }
    cout << cnt + 1 << '\n';
}

int main() {
    int n;
    cin >> n;
    for (int i = 0; i < n; i++)
        f();
}

File: Greedy/Python/1946.py
#  신입사원

def f():
    n = int(input())
    v = []
    for _ in range(n):
        v.append(list(map(int, input().split())))
    v.sort(key=lambda x: [x[0], -x[1]])
    cri = v[0][1]
    cnt = 0
    for i in range(1, n):
        if cri >= v[i][1]:
            cnt += 1
            cri = v[i][1]
    print(cnt+1)


n = int(input())
for _ in range(n):
    f()
